hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -2402,6 +2406,9 @@ int MDMParser::_cbCOPS(int type, const char* buf, int len, NetStatus* status)
             status->cgi.mobile_country_code = static_cast<uint16_t>(::atoi(mobileCountryCode));
             status->cgi.mobile_network_code = static_cast<uint16_t>(::atoi(mobileNetworkCode));
 
+            if( act == -1 ) {
+                act = UBLOX_SARA_RAT_LTE_CAT_M1;
```

Comment:
Its not clear to me if non SARA R modules use these static callbacks. I'm assuming not in this case, or at least that if we see a -1 value then we can assume its a R510 and should be changed

Reply:
We should probably add a check here for R510 and/or R410 like above, since this callback is used for all u-blox modem types we support.

NIT: spacing for `if (`

Reply:
I had the same instinct, but it appears the COPS handler is a static function that is invoked from `waitFinalResp` like so
```
sendFormated("AT+COPS?\r\n");
if (RESP_OK != waitFinalResp(_cbCOPS, &_net, COPS_TIMEOUT)) {
```
```
int MDMParser::waitFinalResp(_CALLBACKPTR cb /* = NULL*/,
                             void* param /* = NULL*/,
                             system_tick_t timeout_ms /*= 10000*/)
{
...
            if (cb) {
                int len = LENGTH(ret);
                int ret = cb(type, buf, len, param);
                if (WAIT != ret)
                    return ret;
            }
...
}
```
`modemIsSaraRxFamily()` relies on the `_dev.dev` class member to check for modem type. I'm not sure how to do this device check from a static context just yet

Reply:
Oh, yeah that is a problem... so perhaps just post process it after these calls:
```
sendFormated("AT+COPS?\r\n");
if (RESP_OK != waitFinalResp(_cbCOPS, &_net, COPS_TIMEOUT)) {
    goto failure;
}
```

Reply:
I can do something like this after each `AT+COPS?` query:
```
            sendFormated("AT+COPS?\r\n");
            if (RESP_OK != waitFinalResp(_cbCOPS, &_net, COPS_TIMEOUT)) {
                goto failure;
            }
            if (_net.act == ACT_UNKNOWN && modemIsSaraRxFamily()){
                _net.act = ACT_LTE_CAT_M1;
            }
```
There are 4 instances in `mdm_hal.cpp` where `AT+COPS?` is sent, should I do this for each? or make another helper function to check and modify `_net.act` after the command is sent from the non-static context?

Reply:
Another helper function sounds appropriate.  I think we should standardize on only making this change for R510 though.  So far we have a few places where the change is made for R510 only and some for R510 & R410.